File: Data Structures & Algorithms/surrounded-regions/submission-1.java
class Solution {
    public void solve(char[][] board) {

        for(int i = 0; i<board.length;i++){
            for(int j =0; j<board[0].length;j++){
                if(board[i][j] == 'O' && (i==0 || i==board.length-1 || j==0 || j==board[0].length-1)){
                    dfs(i,j,board);
                }
            }
        }

        for(int i = 0; i<board.length;i++){
            for(int j =0; j<board[0].length;j++){
                if(board[i][j] == 'O'){
                    board[i][j]= 'X';
                }

            }
        }

        for(int i = 0; i<board.length;i++){
            for(int j =0; j<board[0].length;j++){
                if(board[i][j] == 'H'){
                    board[i][j]= 'O';
                }

            }
        }




        
    }
    public void dfs(int i,int j, char[][] board){
        if(i<0||j<0||i==board.length||j==board[0].length|| board[i][j] !='O'){
            return;
        }
        board[i][j] = 'H';
        dfs(i+1,j,board);
        dfs(i,j+1,board);
        dfs(i,j-1,board);
        dfs(i-1,j,board);
    }
}

File: Data Structures & Algorithms/search-for-word/submission-1.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        rows,cols = len(board),len(board[0])
        path=set()
        def dfs(x,r,c):
            if x == len(word):
                return True
            if (r<0 or c<0  or r>=rows or c>=cols or  word[x] != board[r][c] or (r,c) in path):
                return False
            
            path.add((r,c))
            res= ( dfs(x+1,r+1,c) or 
                    dfs(x+1,r-1,c ) or
                    dfs(x+1,r,c+1 ) or
                    dfs(x+1,r,c-1 )) 
            path.remove((r,c))
            return res




        
        for i in range(len(board)):
            for j in range(len(board[0])):
                if dfs(0,i,j): return True
        return False

